packages/js/test/unit/server/integrations/unhandled_rejection_monitor.server.test.ts

```diff
@@ -0,0 +1,84 @@
+import { TestTransport, TestClient, nullLogger } from '../../helpers'
+import * as util from '../../../../src/server/util'
+import Singleton from '../../../../src/server'
+import UnhandledRejectionMonitor from '../../../../src/server/integrations/unhandled_rejection_monitor'
+
+
+describe('UnhandledRejectionMonitor', () => {
+  // Using any rather than the real type so we can test and spy on private methods
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  let unhandledRejectionMonitor: any
+  let client: typeof Singleton
+  let fatallyLogAndExitSpy: jest.SpyInstance
+  let notifySpy: jest.SpyInstance
+
+  beforeEach(() => {
+    // We just need a really basic client, so ignoring type issues here
+    client = new TestClient(
+      { apiKey: 'testKey', afterUncaught: jest.fn(), logger: nullLogger() }, 
+      new TestTransport()
+    ) as unknown as typeof Singleton
+    unhandledRejectionMonitor = new UnhandledRejectionMonitor(client)
+    // Have to mock fatallyLogAndExit or we will crash the test
+    fatallyLogAndExitSpy = jest
+      .spyOn(util, 'fatallyLogAndExit')
+      .mockImplementation(() => true as never)
+    notifySpy = jest.spyOn(client, 'notify')
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+    process.removeAllListeners('unhandledRejection')
+    unhandledRejectionMonitor.__isReporting = false
+  })
+
+  describe('constructor', () => {
+    it('set up variables and client', () => {
+      expect(unhandledRejectionMonitor.__isReporting).toBe(false)
+      expect(unhandledRejectionMonitor.__client.config.apiKey).toBe('testKey')
+    })
+  })
+
+  describe('handleUnhandledRejection', () => {   
+    const promise = new Promise(() => true)
+    const reason = 'Promise rejection reason'
+
+    it('calls notify and fatallyLogAndExit', (done) => {   
+      unhandledRejectionMonitor.handleUnhandledRejection(reason, promise)
+      expect(notifySpy).toHaveBeenCalledTimes(1)
+      expect(notifySpy).toHaveBeenCalledWith(
+        reason, 
+        { component: 'unhandledRejection' },
+        { afterNotify: expect.any(Function) }
+      )
+      client.afterNotify(() => {
+        expect(fatallyLogAndExitSpy).toHaveBeenCalledWith(reason)
+        done()
+      }) 
+    })
+  
+    // Seems like it was meant to work this way but was never implemented?
```

Comment:
This is easy to add to make it match how `uncaughtException` works (eg return right away if `__isReporting` is `true`... but I'm curious if there's a reason it doesn't include that? Also was trying to make a `chore` PR rather than a `fix` PR.

Reply:
🤔 I don't remember. So, now it was **changed** to return is `__isReporting` is `true`?

Reply:
Here's an issue to review this! https://github.com/honeybadger-io/honeybadger-js/issues/1167